contentcuration/contentcuration/views/channels.py

```diff
@@ -31,24 +29,26 @@ def generate_response(filepath, content_type, filename=None):
     return response
 
 
-@login_required
+@can_access_channel
 @api_view(['GET'])
-@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((AllowAny,))
```

Comment:
I think we can probably stop generating these on the server side at all, and follow the things that Blaine has been doing on Kolibri for generating stuff in the browser, I know there are JS libraries to generate PPT and PDF at the very least.

Reply:
I agree! Maybe as a future pass? I opened an [issue](https://github.com/learningequality/studio/issues/1729) up in the meantime